src/invoke.rs

```diff
@@ -30,10 +41,99 @@ pub enum Error {
     Host(#[from] HostError),
 }
 
+pub struct Snap {
+    storage: OrdMap<LedgerKey, LedgerEntry>,
```

Comment:
This type alone doesn't provide storage, it holds ledger entries to be stored, so I suggest renaming the field:
```suggestion
    ledgerEntries: OrdMap<LedgerKey, LedgerEntry>,
```